### PR TITLE
Issue/3662 add customer list

### LINKED ISSUE #3662: Order creation: Customer loading
The first page of customers should be loaded after opening the screen. A skeleton is shown during the loading. The next page is shown after scrolling to the bottom. If there are no customers, an empty view is shown (see `WCEmptyView`).

image

### PR DESCRIPTION
**Pull request by kidinov**
Resolves #3662

Add customer screen with a list of customers

**Testing**
* Go to the orders screen
* Click FAB
* Click "START CUSTOMER CREATION"
* Add customer screen should be visible with customers

Video:

https://user-images.githubusercontent.com/4923871/110446956-cb761b00-80d0-11eb-8ba0-9b93a54a9907.mp4

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -82,7 +82,7 @@ task installGitHooks(type: Copy) {
 }
 
 ext {
-    fluxCVersion = '1.11.0'
+    fluxCVersion = '982abdc398f957b5490a514df9fa58f237c066bd'
```
Comment (kidinov) [inline build.gradle:85]: This change should be reverted probably before merging? Without this, it won't pass any CI checks

File: libs/login/WordPressLoginFlow/build.gradle [hunk 2]
```diff
@@ -60,12 +60,12 @@ dependencies {
 
     // Share FluxC version from host project if defined, otherwise fallback to default
     if (project.hasProperty("fluxCVersion")) {
-        implementation("com.github.wordpress-mobile.WordPress-FluxC-Android:fluxc:$fluxCVersion") {
+        implementation("com.github.kidinov.WordPress-FluxC-Android:fluxc:$fluxCVersion") {
```
Comment (kidinov) [inline libs/login/WordPressLoginFlow/build.gradle:63]: Same here https://github.com/woocommerce/woocommerce-android/pull/3671/files#r590112917

File: libs/login/WordPressLoginFlow/build.gradle [hunk 3]
```diff
@@ -60,12 +60,12 @@ dependencies {
...
             exclude group: "com.android.support"
             exclude group: "org.wordpress", module: "utils"
         }
     } else {
-        implementation("com.github.wordpress-mobile.WordPress-FluxC-Android:fluxc:1.6.22") {
+        implementation("com.github.kidinov.WordPress-FluxC-Android:fluxc:1.6.22") {
```
Comment (kidinov) [inline libs/login/WordPressLoginFlow/build.gradle:68]: Same here https://github.com/woocommerce/woocommerce-android/pull/3671/files#r590112917

File: settings.gradle [hunk 4]
```diff
@@ -1,7 +1,7 @@
 include ':WooCommerce'
 include ':libs:login:WordPressLoginFlow'
 
-gradle.ext.fluxCBinaryPath = "com.github.wordpress-mobile.WordPress-FluxC-Android"
+gradle.ext.fluxCBinaryPath = "com.github.kidinov.WordPress-FluxC-Android"
```
Comment (kidinov) [inline settings.gradle:4]: Same here https://github.com/woocommerce/woocommerce-android/pull/3671/files#r590112917

File: WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/WCEmptyView.kt [hunk 5]
```diff
@@ -182,6 +186,20 @@ class WCEmptyView @JvmOverloads constructor(ctx: Context, attrs: AttributeSet? =
...
+                message = null
+                buttonText = null
+                drawableId = R.drawable.img_empty_orders_no_orders
+            }
+            CUSTOMER_LIST_LOADING -> {
+                isTitleBold = true
+                title = context.getString(R.string.order_creation_add_customer_empty_list_loading_title)
+                message = null
+                buttonText = null
+                drawableId = R.drawable.img_empty_orders_loading
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/WCEmptyView.kt:203]: I use images that were used for the orders. They look ok, but I don't see any design in Figma for this state, so not sure if it's ok

Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/WCEmptyView.kt:203]: That's fine. Maybe just add a TODO for now.

File: WooCommerce/src/main/res/layout/fragment_order_creation_add_customer.xml [hunk 6]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (0nko) [inline WooCommerce/src/main/res/layout/fragment_order_creation_add_customer.xml]: You can use `com.woocommerce.android.widgets.WCElevatedConstraintLayout` here and it'll style the view with the correct background color.

File: WooCommerce/src/main/res/layout/fragment_order_creation_new_order.xml [hunk 7]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (0nko) [inline WooCommerce/src/main/res/layout/fragment_order_creation_new_order.xml]: Same here.

**Review by 0nko — COMMENTED** [review]
I tried to change the role parameter to `all` but the list is then stuck loading (the first item was already loaded with the original default value):

image

File: WooCommerce/src/main/res/layout/fragment_order_creation_new_order.xml (hunk 7 above)
Comment (kidinov) [reply WooCommerce/src/main/res/layout/fragment_order_creation_new_order.xml]: [Addressed](https://github.com/woocommerce/woocommerce-android/pull/3671/commits/c663addf8789a5adcfdb12cb70700feaf7a237a8)

File: WooCommerce/src/main/res/layout/fragment_order_creation_add_customer.xml (hunk 6 above)
Comment (kidinov) [reply WooCommerce/src/main/res/layout/fragment_order_creation_add_customer.xml]: [Addressed](https://github.com/woocommerce/woocommerce-android/pull/3671/commits/c663addf8789a5adcfdb12cb70700feaf7a237a8)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/WCEmptyView.kt (hunk 5 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/WCEmptyView.kt:203]: [Addressed](https://github.com/woocommerce/woocommerce-android/pull/3671/commits/c663addf8789a5adcfdb12cb70700feaf7a237a8)

**Review by 0nko — APPROVED** [review]
:shipit:

**Pull request merged**